magda/config_reader.py

```
@@ -21,10 +26,18 @@ async def read(
         cls,
         config: str,
         module_factory: ModuleFactory,
+        config_parameters: Optional[Dict] = None,
         context: Optional[Any] = None,
-        shared_parameters: Optional[Dict] = None,
+        shared_parameters: Optional[Dict] = None
     ):
+        if config_parameters:
+            cls._validate_config_parameters_structure(config_parameters)
```

What about the case, when `config_parameters` are not provided but config has parameters? (it should raise an exception)

---

My bad :) Good that you noticed. Will be refactored